src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -51,19 +35,18 @@ const HeatmapMetadataTrackSettings = (props) => {
     trackData,
     'key',
   );
-
   useEffect(() => {
     // Prevent initial dispatch when object appears
-    if (isInitialRenderRef.current) {
+    if (isInitialRenderRef.current
+      || _.isEqual(selectedHierarchy, currentSelectedHierarchy.current)) {
```

Comment:
For some reason `selectedHierarchy` would trigger this useEffect even though the values remain the same. This is why I check if they are equal to a copy here. It's weird, maybe it has to do with the initialization of the variabla every time? I can see in the logs that getCellsSetsHierarchy is called every time, or the` .filter` `.map` that we do on `selectedHierarchy`. Not sure how this happens.

Reply:
that's weird `hierarchy` triggered this useEffect at those times too, right?

Reply:
it might be that the reference changes, that might cause `useEffect` to trigger again, in that case if we used memoizing it would keep referential equality

Reply:
I was actually looking into this issue already and found this library: https://github.com/reduxjs/reselect.
I was planning to add a ticket to do this in the future, but perhaps you think it is worth a try to deal with this?